modules/core/src/model/PurposeRestrictionVector.ts

```diff
@@ -148,11 +148,13 @@ export class PurposeRestrictionVector extends Cloneable<PurposeRestrictionVector
 
     })();
 
+    const values = [...Array(lastEntry).keys()].map( (i) => i + 1);
+
     for (let i = 1; i <= lastEntry; i++) {
 
       if (!this.has(hash)) {
 
-        this.map.set(hash, new BinarySearchTree());
+        this.map.set(hash, BinarySearchTree.build(values));
```

Comment:
An instance of `BinarySearchTree` is also used in line 116 in the `add` function. Perhaps a comment here on why this bst is created using the static method.

Reply:
Yes, I was thinking what the best way to implement the `build` function, one option is using constructor but have decided to implement it as a static method as it looks like it very specific way to create `BinarySearchTree` from the ordered array. I have added a comment in the line, thanks